From 51324698a88520c835cf94b77723f7c4792f0e04 Mon Sep 17 00:00:00 2001
From: Robert Marko <robimarko@gmail.com>
Date: Wed, 5 Sep 2018 12:01:23 +0200
Subject: [PATCH] rpi-3b-plus: Enable ethernet LEDs

Enable the ethernet LEDs on Pi 3B+ to work since driver support
was backported from 4.18 previously

Signed-off-by: Robert Marko <robimarko@gmail.com>
---
 arch/arm/boot/dts/bcm283x-rpi-lan7515.dtsi | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/arch/arm/boot/dts/bcm283x-rpi-lan7515.dtsi b/arch/arm/boot/dts/bcm283x-rpi-lan7515.dtsi
index 146ef801..30e47251 100644
--- a/arch/arm/boot/dts/bcm283x-rpi-lan7515.dtsi
+++ b/arch/arm/boot/dts/bcm283x-rpi-lan7515.dtsi
@@ -1,3 +1,5 @@
+#include <dt-bindings/net/microchip-lan78xx.h>
+
 / {
       aliases {
               ethernet = &ethernet;
@@ -20,8 +22,21 @@
                       ethernet: usbether@1 {
                               compatible = "usb424,7800";
                               reg = <1>;
+
                               microchip,eee-enabled;
                               microchip,tx-lpi-timer = <600>; /* non-aggressive*/
+
+                              mdio {
+                                      #address-cells = <0x1>;
+                                      #size-cells = <0x0>;
+                                      eth_phy: ethernet-phy@1 {
+                                              reg = <1>;
+                                              microchip,led-modes = <
+                                                      LAN78XX_LINK_1000_ACTIVITY
+                                                      LAN78XX_LINK_10_100_ACTIVITY
+                                              >;
+                                      };
+                              };
                       };
               };
       };
-- 
2.17.1

